src/components/Form/Form.tsx

```diff
@@ -0,0 +1,128 @@
+  content?: ShorthandValue
+  onSubmit?: ComponentEventHandler<IFormProps>
+  fields?: ShorthandValue[]
+  renderField?: ShorthandRenderFunction
+  styles?: ComponentPartStyle
+  variables?: ComponentVariablesInput
+}
+
+/**
+ * A Form displays a set of related user input fields in a structured way.
+ * @accessibility
+ * Label needs to be provided by using 'aria-label', or 'aria-labelledby' attributes on the <form> element.
+ */
+class Form extends UIComponent<Extendable<IFormProps>, any> {
+  static create: Function
+
+  public static displayName = 'Form'
+
+  public static className = 'ui-form'
+
+  public static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /**
+     *  Form content for childrenApi
+     *  @docSiteIgnore
+     */
+    children: PropTypes.node,
+
+    /** Additional CSS class name(s) to apply.  */
+    className: PropTypes.string,
+
+    /** Shorthand for primary content. */
+    content: customPropTypes.contentShorthand,
+
+    /** Shorthand array of props for the Form.Fields inside the Form. */
+    fields: customPropTypes.collectionShorthand,
+
+    /**
+     * A custom render iterator for rendering each of the Form fields.
+     * The default component, props, and children are available for each field.
+     *
+     * @param {React.ReactType} Component - The computed component for this slot.
+     * @param {object} props - The computed props for this slot.
+     * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+     */
+    renderField: PropTypes.func,
+
+    /**
+     * The HTML form submit handler.
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onSubmit: PropTypes.func,
+
+    /** The HTML form action. */
+    action: PropTypes.string,
+
+    /** Additional CSS styles to apply to the component instance.  */
+    styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Override for theme site variables to allow modifications of component styling via themes. */
+    variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+  }
+
+  public static defaultProps = {
+    as: 'form',
+  }
+
+  public static Field = FormField
+
+  public renderComponent({
+    ElementType,
+    classes,
+    accessibility,
+    variables,
+    styles,
+    rest,
+  }): React.ReactNode {
+    const { action, children } = this.props
+    return (
+      <ElementType className={classes.root} action={action} onSubmit={this.handleSubmit} {...rest}>
+        {childrenExist(children) ? children : this.renderFields()}
+      </ElementType>
+    )
+  }
+
+  private handleSubmit = (e, ...args) => {
+    const { action } = this.props
+
+    // Heads up! Third party libs can pass own data as first argument, we need to check that it has preventDefault()
+    // method.
+    if (typeof action !== 'string') _.invoke(e, 'preventDefault')
```

Comment:
could you, please, elaborate on the case we are trying to address here? How it is read by me is:
- invoke `onSubmit` props callback for the proper set of arguments being provided when form is submitted
- but if `action` is not a string, call `preventDefault` on the event object

Frankly, a bit confused with the second condition. What we are trying to handle with that? And, suppose that I am passing some garbish but `string` value as action, like `aaa` - shouldn't we call `preventDefault` in that case as well?

Reply:
This was taken from Semantic UI. Essentially, the onSubmit should be invoke only if the action is valid, in other cases the users usually use the onClick on some button inside the Form, that's why we are preventing this.

Reply:
This is taken from Semantic UI. The explanation for this is in the comment above.

Reply:
yes, I've got it - but still cannot get the idea from the explanation, could you, please, help me with that? Specifically,  what am confused about is the following:

> a bit confused with the second condition. What we are trying to handle with that? And, suppose that I am passing some garbish but string value as action, like aaa - shouldn't we call preventDefault in that case as well?